setup.py

```diff
@@ -35,14 +37,91 @@ def is_conda_interpreter():
     """
     return 'conda' in sys.version or 'Continuum' in sys.version
 
+def is_macOS():
+    return 'darwin' in sys.platform
+
+
+# HOMEBREW_SAMPLE = """$ brew ls --verbose libomp
+# /opt/homebrew/Cellar/libomp/15.0.7/INSTALL_RECEIPT.json
+# /opt/homebrew/Cellar/libomp/15.0.7/.brew/libomp.rb
+# /opt/homebrew/Cellar/libomp/15.0.7/include/ompt.h
+# /opt/homebrew/Cellar/libomp/15.0.7/include/omp.h
+# /opt/homebrew/Cellar/libomp/15.0.7/include/omp-tools.h
+# /opt/homebrew/Cellar/libomp/15.0.7/lib/libomp.dylib
+# /opt/homebrew/Cellar/libomp/15.0.7/lib/libomp.a
+# """
+#
+# MACPORTS_SAMPLE = """$ port contents libomp
+# Port libomp contains:
+#   /opt/local/include/libomp/omp-tools.h
+#   /opt/local/include/libomp/omp.h
+#   /opt/local/include/libomp/ompt.h
+#   /opt/local/lib/libomp/libgomp.dylib
+#   /opt/local/lib/libomp/libiomp5.dylib
+#   /opt/local/lib/libomp/libomp.dylib
+#   /opt/local/share/doc/libomp/LICENSE.TXT
+#   /opt/local/share/doc/libomp/README.txt
+# """
+
+
+def compile_link_paths_from_manifest(cmd):
+    from subprocess import run
+    query = run(cmd, shell=True, check=False, capture_output=True)
+    if query.returncode != 0:
+        return None, None
+    manifest = query.stdout.decode("UTF-8")
+    # find all the unique directories mentioned in the manifest
+    dirs = set(os.path.split(filename)[0] for filename in re.findall(r'^\s*(/.*?)\s*$', manifest, re.MULTILINE))
+    # find a unique libdir and incdir
+    inc = tuple(d for d in dirs if re.search(r'/include(\W|$)', d))
+    lib = tuple(d for d in dirs if re.search(r'/lib(\W|$)', d))
+    # only return success if there's no ambiguity
+    return (inc + lib) if len(inc) == 1 and len(lib) == 1 else (None, None)
+
+
+@lru_cache(maxsize=1)
+def macOS_omp_options_from_probe():
+    """Get common include and library paths for libomp installation on macOS.
+       For example ``(['-I/opt/local/include/libomp'], ['-L/opt/local/lib/libomp'])``.
+    """
+    if is_macOS():
+        for cmd in ["brew ls --verbose libomp", "port contents libomp"]:
+            inc, lib = compile_link_paths_from_manifest(cmd)
+            if inc and lib:
+                return [f"-I{inc}"], [f"-L{lib}"]
+    return [], []
+
+def is_macOS_with_libomp():
+    if not is_macOS():
+        return False
+    inc, lib = macOS_omp_options_from_probe()
+    return bool(inc) and bool(lib)
+
+OMP_TAB = {
+    '1': 'gomp',
+    '0': None,
+    'gcc': 'gomp',
+    'gomp': 'gomp',
+    'clang': 'omp',
+    'omp': 'omp',
+    'probe': ('gomp' if (not is_macOS() or is_conda_interpreter())
+              else ('omp' if is_macOS_with_libomp() else None))
```

Comment:
Can we make the value here also be `probe` and then make `'1'` map to `probe` as well, then in the handling function we can have as much complicated logic as needed and it won't have to be combined into this small one liner *and* it won't run at import/load time. It should also mean you can remove the `lrucache` on `macOS_omp_options_from_probe` which screams to me a need for a refactor.

Reply:
Agreed that moving probing actions out of import-time would be an improvement. Tipping things over into class members or a new helper class is not necessarily more approachable to folks just trying to get a build, but I'll have a look.

Reply:
Most recent push consolidates the OpenMP probing to only happen when user has left it up to the program to figure it out (now the default case). Also moved the msvc /openmp option into the argument tables and compiler is now a parameter in the OpenMP _omp_compile_link_args member function of the build_ext_subclass. Tested by hand with these options on my M2 mac with macports with a python 3.11 venv test setup:

```
pip install -v .
USE_OMP=0 pip install -v . 
CC=gcc-mp-12 USE_OMP=gcc pip install -v . 
USE_OMP=clang pip install -v .
```

verification done with checking for _omp_ (GCC) or _.omp_ symbols (Clang) in the dynamic lib:
```
nm build/lib.macosx-13.0-arm64-cpython-311/pykdtree/kdtree.cpython-311-darwin.so |grep omp
```

Also ran the test_tree in a separate directory to avoid confusion about local source directory
```
python -m pytest test_tree.py
```

Between each test, reset is done with
```
pip uninstall pykdtree; rm -fr build
```